plugin.audio.ampache/resources/lib/ampache_plugin.py

```diff
@@ -0,0 +1,1053 @@
+    disknumber = str(node.findtext("disk"))
+    if PY2:
+        fullname = str(node.findtext("name"))
+        fullname += u" - "
+        fullname += str(node.findtext("artist"))
+    else:
+        #no encode utf-8 in python3, not necessary
+        fullname = node.findtext("name")
+        fullname += " - "
+        fullname += node.findtext("artist")
+    #disknumber = "None" when disk number is not sent
+    if disknumber!="None" and disknumber != "1" and disknumber !="0":
+        if PY2:
+            fullname = fullname + u" - [ " + ut.tString(30195) + u" " +\
+               disknumber + u" ]"
+        else:
+            fullname = fullname + " - [ " + ut.tString(30195) + " " + disknumber + " ]"
+    return fullname
+
+def getRating(rating):
+    if rating:
+        #converts from five stats ampache rating to ten stars kodi rating
+        rating = int(float(rating)*2)
+    else:
+        #zero equals no rating
+        rating = 0
+    return rating
+
+def get_infolabels(object_type , node):
+    infoLabels = None
+    rating = getRating(node.findtext("rating"))
+    if object_type == 'albums':
+        infoLabels = {
+            'Title' : str(node.findtext("name")) ,
+            'Album' : str(node.findtext("name")) ,
+            'Artist' : str(node.findtext("artist")),
+            'DiscNumber' : str(node.findtext("disk")),
+            'Year' : node.findtext("year") ,
+            'UserRating' : rating,
+            'Mediatype' : 'album'
+        }
+ 
+    elif object_type == 'artists':
+        infoLabels = {
+            'Title' : str(node.findtext("name")) ,
+            'Artist' : str(node.findtext("name")),
+            'Mediatype' : 'artist'
+        }
+
+    elif object_type == 'songs':
+        infoLabels = {
+            'Title' : str(node.findtext("title")) ,
+            'Artist' : str(node.findtext("artist")),
+            'Album' :  str(node.findtext("album")),
+            'Size' : node.findtext("size") ,
+            'Duration' : node.findtext("time"),
+            'Year' : node.findtext("year") ,
+            'TrackNumber' : node.findtext("track"),
+            'UserRating' : rating,
+            'Mediatype' : 'song'
+        }
+
+    return infoLabels
+
+#this function is used to speed up the loading of the images using differents
+#theads, one for request
+def precacheArt(elem,object_type):
+    elem_type = ut.otype_to_type(object_type)
+    if elem_type != "album" or elem_type != "song":
+        return
+    art_type = "album"
+    for node in elem.iter(elem_type):
+        if elem_type == "song":
+            try:
+                album_elem = node.find("album")
+                object_id = int(album_elem.attrib["id"])
+            except:
+                object_id = None
+        else:
+            object_id = int(node.attrib["id"])
+        image_url = check_get_art_url(node)
+        x = threading.Thread(target=art.get_art,args=(object_id,art_type,image_url,))
+        x.start()
+    x.join()
```

Comment:
I don't think this is right, X is a new thread instance each time you iterate on the for loop. You're only waiting for the last one to finish and not all of them. Save a thread reference into a list and execute join on all of them after the loop ends